### PR TITLE
Change court report select text transitioning to be calculated from birthdate

### LINKED ISSUE #2343: Change court report select text transitioning to be calculated from birthdate
### Impacted User Types
 - volunteers
 - supervisors
 - admins

### Environment
all

### Current Behavior
Casa cases where the birthdate occurred more than 14 years ago may not be marked as transitioning
As an admin, https://casa-qa.herokuapp.com/case_court_reports will take you to the relevant page.
Transition Aged Incorrectly Labelled

### Expected Behavior
Casa cases where the birthdate occurred more than 14 years ago should be marked as transitioning

see `app/decorators/casa_case_decorator.rb`

### PR DESCRIPTION
**Pull request by carolyn-manning**
Updated files to use birthday rather than transition_aged_youth boolean to calculate whether casa case is transition aged youth

### What github issue is this PR for, if any?
Resolves #2343 

### What changed, and why?
Updated certain instances of the transitioned_aged_youth attribute to calculate if the casa case is under 14 years old from birthday instead. 

### How is this tested? (please write tests!) 💖💪

These tests are still passing. A few tests are now failing since they are still using the boolean value. I am working on cleaning those test/everything else still using or defaulting to that boolean up still though! It is only being referenced in a few more places - mainly specs. 

it "returns transition age youth status with icon if over 14 years old" do
      casa_case = build(:casa_case, birth_month_year_youth: 14.years.ago)
      expect(casa_case.decorate.transition_aged_youth)
        .to eq "Yes #{CasaCase::TRANSITION_AGE_YOUTH_ICON}"
    end

    it "returns non-transition age youth status with icon if not over 14 years old" do
      casa_case = build(:casa_case, birth_month_year_youth: 13.years.ago)
      expect(casa_case.decorate.transition_aged_youth)
        .to eq "No #{CasaCase::NON_TRANSITION_AGE_YOUTH_ICON}"
    end
  end

### Screenshots please :)
Screen Shot 2022-08-19 at 6 43 31 PM

Im happy to keep working on this and create a new pull request once the transition_aged_youth column in the db is deleted if easier!

### CODE REVIEW

**Comment by compwron** [comment]
This is very cool! How are you feeling about this?

**Comment by carolyn-manning** [comment]
I feel good about this. I was looking into cleaning it up a bit though and was thinking about defining a scope in the casa case model (app/models/casa_case.rb) and using that instead of the. where("birth_month_year_youth < ?", 14.years.ago) statement repeatedly. Do you think that would be better?

**Comment by compwron** [comment]
> I feel good about this. I was looking into cleaning it up a bit though and was thinking about defining a scope in the casa case model (app/models/casa_case.rb) and using that instead of the. where("birth_month_year_youth < ?", 14.years.ago) statement repeatedly. Do you think that would be better?

making a new scope sounds like a great way to de-duplicate some code, I am for it :)

File: app/controllers/emancipation_checklists_controller.rb [hunk 1]
```diff
@@ -1,11 +1,12 @@
 class EmancipationChecklistsController < ApplicationController
+  include DateHelper
   before_action :require_organization!
   after_action :verify_authorized
 
   def index
     authorize :application, :see_emancipation_checklist?
     org_cases = current_user.casa_org.casa_cases.includes(:assigned_volunteers)
-    @casa_transitioning_cases = policy_scope(org_cases).where(transition_aged_youth: true).includes([:hearing_type, :judge])
+    @casa_transitioning_cases = policy_scope(org_cases).where("birth_month_year_youth < ?", 14.years.ago).includes([:hearing_type, :judge])
```
Comment (italomatos) [inline app/controllers/emancipation_checklists_controller.rb:9]: Hi @carolyn-manning! how are you? , there is a good way to do this using infinity range.

```
  @casa_transitioning_cases = policy_scope(org_cases).where(birth_month_year_youth: 14.years.ago..).includes([:hearing_type, :judge])
```

File: app/models/casa_case.rb [hunk 2]
```diff
@@ -152,7 +152,7 @@ def has_judge_name?
   end
 
   def has_transitioned?
-    transition_aged_youth
+    birth_month_year_youth && birth_month_year_youth < 14.years.ago
```
Comment (italomatos) [inline app/models/casa_case.rb:155]: this method its similar this other one ( line 134)
```
def in_transition_age? 
```

File: app/models/user.rb [hunk 3]
```diff
@@ -131,10 +132,9 @@ def active_for_authentication?
   end
 
   def serving_transition_aged_youth?
-    actively_assigned_and_active_cases.where(transition_aged_youth: true).any?
+    actively_assigned_and_active_cases.where("birth_month_year_youth <= ?", 14.years.ago).any?
```
Comment (italomatos) [inline app/models/user.rb:135]: here
```ruby
actively_assigned_and_active_cases
  .where(birth_month_year_youth:, 14.years.ago..).any?
```

File: app/models/casa_case.rb [hunk 4]
```diff
@@ -70,8 +71,7 @@ class CasaCase < ApplicationRecord
       .order(:case_number)
   }
   scope :should_transition, -> {
-    where(transition_aged_youth: false)
-      .where("birth_month_year_youth <= ?", 14.years.ago)
+    where("birth_month_year_youth <= ?", 14.years.ago)
```
Comment (italomatos) [inline app/models/casa_case.rb:74]: suggestion here:
```
where(birth_month_year_youth: 14.years.ago..)
```

**Comment by FireLemons** [comment]
:1st_place_medal:

**Pull request merged**